registration/include/pcl/registration/impl/sample_consensus_prerejective.hpp

```diff
@@ -257,37 +273,37 @@ SampleConsensusPrerejective<PointSource, PointTarget, FeatureT>::computeTransfor
     // Apply prerejection
     if (!correspondence_rejector_poly_->thresholdPolygon(sample_indices,
                                                          corresponding_indices)) {
-      ++num_rejections;
+      num_rejections += 1;
       continue;
     }
 
-    // Estimate the transform from the correspondences, write to transformation_
+    // Estimate the transform from the correspondences, write to transformation
+    Eigen::Matrix4f transformation;
     transformation_estimation_->estimateRigidTransformation(
-        *input_, sample_indices, *target_, corresponding_indices, transformation_);
-
-    // Take a backup of previous result
-    const Matrix4 final_transformation_prev = final_transformation_;
-
-    // Set final result to current transformation
-    final_transformation_ = transformation_;
+    	*input_, sample_indices, *target_, corresponding_indices, transformation);
 
-    // Transform the input and compute the error (uses input_ and final_transformation_)
-    getFitness(inliers, error);
-
-    // Restore previous result
-    final_transformation_ = final_transformation_prev;
+    // Transform the input and compute the error (uses input_ and transformation)
+    float error = getFitness(transformation, inliers);
 
     // If the new fit is better, update results
-    inlier_fraction =
-        static_cast<float>(inliers.size()) / static_cast<float>(input_->size());
-
-    // Update result if pose hypothesis is better
-    if (inlier_fraction >= inlier_fraction_ && error < lowest_error) {
-      inliers_ = inliers;
-      lowest_error = error;
-      converged_ = true;
-      final_transformation_ = transformation_;
+    const float inlier_fraction =
+        static_cast<float> (inliers.size ()) / static_cast<float> (input_->size ());
+
+    if (inlier_fraction >= inlier_fraction_) {
+      #pragma omp critical
+      {
+        // Update result if pose hypothesis is better
+        if (error < lowest_error)
+        {
+          inliers_ = inliers;
+          lowest_error = error;
+          converged_ = true;
+          transformation_ = transformation;
+          final_transformation_ = transformation;
```

Comment:
Off-topic: maybe I'm missing something in the code, but I don't see how `transformation_` and `final_transformation_` differ

Reply:
```transformation_``` is the estimate of the transformation at the last registration iteration. I'm not really sure how it is used, but I guess it was made for registration visualization that's been no longer used.

https://github.com/PointCloudLibrary/pcl/blob/0a425d3f12a9bd1b8b4ceac2d8f6678e6a586840/registration/include/pcl/registration/registration.h#L584-L586

https://github.com/PointCloudLibrary/pcl/blob/0a425d3f12a9bd1b8b4ceac2d8f6678e6a586840/registration/include/pcl/registration/registration.h#L588-L589